File: management-domain/Sales-department/Backend/Node.js/sales-automation-service/src/shared/exceptions/not-found.exception.ts
import { BaseException } from './base.exception';
import { HttpStatus } from '@nestjs/common';

export class NotFoundException extends BaseException {
  readonly statusCode = HttpStatus.NOT_FOUND;
  readonly details: any;

  constructor(message: string, details?: any) {
    super(message);
    this.details = details;
    Object.setPrototypeOf(this, NotFoundException.prototype);
  }
}

File: management-domain/Executive-domain-X/Executive-domain-X/Backend/Java/executive-dashboard-service/Shared/executive-analytics-service/src/main/java/com/gogidix/management/shared/exception/NotFoundException.java
package com.gogidix.management.shared.exception;

public class NotFoundException extends DomainException {
    public NotFoundException(String message) {
        super(message, "NOT_FOUND");
    }

    public NotFoundException(String type, String id) {
        super(type + " not found: " + id, "NOT_FOUND");
    }

    public NotFoundException(String message, Throwable cause) {
        super(message, cause);
    }
}
